### ISSUE TITLE
How to Call Bound Function which returns Collection object

### ISSUE THREAD

**Issue by RavindraBarapatre** [object/Simple.OData.Client#191]
Hello.
I have Bound Function which returns `List<UserPermissions>.`

_webapi.config_
 ```
var PermissionCollection = builder.EntityType<Permission>().Collection;
 var getPermission = PermissionCollection.Function("GetPermissions")
            .ReturnsCollectionFromEntitySet<UserPermissions>("UserPermissions");
            getPermissionByUserAndResource.Parameter<string>("userName");
            getPermissionByUserAndResource.Parameter<string>("resourceName");
```

_Client Query_
`await client.For<Permission>().Function("GetPermissions").Set(new { userName = userName, resourceName = resourceName }).ExecuteAsArrayAsync<UserPermissions>();`

When I run it service Function _GetPermissions_ called successfully and it returns result. but in client it throwing error **Error occured in Web API : Object must implement IConvertible.**

Can you please check, what I did wrong

**Comment by object** [comment]
How is UserPermissions defined?

**Comment by RavindraBarapatre** [comment]
public class UserPermissionSet
    {
        public bool Delete { get; set; }
        public Guid Id { get; set; }
        public bool Insert { get; set; }
        public bool Read { get; set; }
        public Guid ResourceId { get; set; }
        public string ResourceName { get; set; }
        public Guid RoleId { get; set; }
        public bool Update { get; set; }
        public Guid UserId { get; set; }
        public string UserName { get; set; }
    }

**Comment by object** [comment]
Thanks I will investigate this.

**Comment by object** [comment]
Can you try ExecuteAsEnumerableAsync instead of ExecuteAsArrayAsync and see what happens?

**Comment by object** [comment]
Sorry, nope this won't work, I will investigate it further.

**Comment by RavindraBarapatre** [comment]
ExecuteAsEnumerableAsync will return IEnumerable<Permission> instead IList<UserPermissions>.
It won't support

**Comment by object** [comment]
Yes it's a bug.

**Comment by object** [comment]
I made a fix and released version 4.14.1. Can you check it out?

**Comment by RavindraBarapatre** [comment]
sure. :+1: 
Thanks
